SCClassLibrary/Common/GUI/Base/QWindow.sc

```diff
@@ -104,7 +104,16 @@ Window {
 		resizable = true,
 		border = true,
 		server,
-		scroll = false;
+		scroll = false,
+		forceCreate = false;
+
+		if( forceCreate.not, {
+			var existingWindows = Window.allWindows.select({|w| w.name == name});
+			if( existingWindows.size > 0 ) {
+				"Window \"%\" already exists".format(name).warn;
```

Comment:
if you have `forceCreate = false` you don't want warnings, I think.

Reply:
Depends on if `forceCreate = false` gets default because this would prevent windows form popping up. Indicating this behavior, the prevention of a window creation, to the user via a warning is indeed intended.

Reply:
>"Window \"%\" already exists".format(name).warn;

_It is never good to warn when there is no danger._
We should assume that it was intentional if `reuseExisting` is set to true, right?

Better:
`"Reusing existing Window \"%\"".format(name).postln`

Reply:
I think the warning is justified as it marks that a statement of a user is changed in the background to some different behavior (e.g. calling `s.meter()` twice is probably an antipattern and therefore good for a warning - in case this was intended that the statement won't get executed is also worth a warning). It also depends where one is coming from, [in e.g. JS there is a trend](https://github.com/xojs/xo) to stop execution if the linter finds unformatted code or unused variables so every warning stops your program (the *nürnberger trichter* of coding? but at the same time I found this extremely productive once you do not have to manage coding styles yourself anymore. one step beyond this is maybe [vigil](https://github.com/munificent/vigil)).

I am totally on the side for more warnings in SC as I personally had many moments of frustration caused by the muteness of some SC internals - maybe the proper angle could be on how warnings should be obeyed.

In general I think SC could really benefit form some kind of advanced logging mechanism (see e.g. https://docs.python.org/3/howto/logging.html#logging-flow which is a good basic logging workflow for a standard library IMO) which could be adjusted depending on the situation or liking of anybody.